README.md

```diff
@@ -63,76 +72,91 @@ python -m pip install --upgrade oneagent-sdk
 
 <a name="initialization-and-sdk-objects"></a>
 ### Initialization and SDK objects
 
 Before first using any other SDK functions, you need to initialize the SDK.
 
 ```python
-init_result = oneagent.try_init()
+init_result = oneagent.initialize()
 print('OneAgent SDK initialization result' + repr(init_result))
 if init_result:
     print('SDK should work (but agent might be inactive).')
 if not init_result:
     print('SDK will definitely not work (i.e. functions will be no-ops).')
 ```
 
-See the API documentation for the [`try_init` function](https://dynatrace.github.io/OneAgent-SDK-for-Python/docs/sdkref.html#oneagent.try_init) and the [`InitResult` class](https://dynatrace.github.io/OneAgent-SDK-for-Python/docs/sdkref.html#oneagent.InitResult) for more information.
+See the API documentation for the [`initialize` function](https://dynatrace.github.io/OneAgent-SDK-for-Python/docs/sdkref.html#oneagent.initialize)
+and the [`InitResult` class](https://dynatrace.github.io/OneAgent-SDK-for-Python/docs/sdkref.html#oneagent.InitResult) for more information.
 
 To use the SDK, get a reference to the [`SDK`](https://dynatrace.github.io/OneAgent-SDK-for-Python/docs/sdkref.html#oneagent.sdk.SDK)
-singleton by calling its static [`get`](https://dynatrace.github.io/OneAgent-SDK-for-Python/docs/sdkref.html#oneagent.sdk.SDK.get)
+singleton by calling the oneagent static [`get_sdk`](https://dynatrace.github.io/OneAgent-SDK-for-Python/docs/sdkref.html#oneagent.get_sdk)
 method. The first thing you may want to do with this object, is checking if the agent is active by comparing the value of the
 [`agent_state`](https://dynatrace.github.io/OneAgent-SDK-for-Python/docs/sdkref.html#oneagent.sdk.SDK.agent_state)
-property to the [`oneagent.common.AgentState`](https://dynatrace.github.io/OneAgent-SDK-for-Python/docs/sdkref.html#oneagent.common.AgentState) constants.
+property to the [`oneagent.common.AgentState`](https://dynatrace.github.io/OneAgent-SDK-for-Python/docs/sdkref.html#oneagent.common.AgentState)
+constants. You can also have a look at the [extended SDK state information](#extended-sdk-state).
 
 ```python
-import oneagent.sdk
+import oneagent
 from oneagent.common import AgentState
 # Initialize oneagent, as above
 
-sdk = oneagent.sdk.SDK.get()
-if sdk.agent_state not in (AgentState.ACTIVE,
-        AgentState.TEMPORARILY_INACTIVE):
+sdk = oneagent.get_sdk()
+if sdk.agent_state not in (AgentState.ACTIVE, AgentState.TEMPORARILY_INACTIVE):
     print('Too bad, you will not see data from this process.')
 ```
 
+As a development and debugging aid it is recommended to set a diagnostic callback. The callback will be used by the SDK to inform about unusual events.
+
+Unusual events that prevent an operation from completing successfully include:
+* API usage errors
+* other unexpected events (like out of memory situations)
+
+The exact situations and arguments in/with which the callback is invoked are undocumented and may change at any time.
```

Comment:
I would remove this sentence. It feels strange that we are explicitly telling the user that we are not documenting something.

Reply:
fixed

Reply:
Then maybe "undefined" instead of "undocumented"? In any case, should be changed at https://dynatrace.github.io/OneAgent-SDK-for-Python/docs/sdkref.html#oneagent.sdk.SDK.set_diagnostic_callback too.

Reply:
I like the warning in https://dynatrace.github.io/OneAgent-SDK-for-Python/docs/sdkref.html#oneagent.sdk.SDK.agent_version_string

a similar warning can be used in https://dynatrace.github.io/OneAgent-SDK-for-Python/docs/sdkref.html#oneagent.sdk.SDK.set_diagnostic_callback

Reply:
What about:
"Use this as a development and debugging aid only. Your application should not rely on a calling sequence or any arguments being set or passed to the callback."